CrisisTextLine/Elmer
# Drop melted update

#### What's this PR do?
Refactors the melted user script to:
  search for melted users only inactive course
  track melted state using CTL online role
  update roles in CTL online correctly
Expand CTLonline API capabilities
Simplify some of removeChurned
#### Where should the reviewer start?
jobs/training/dropMeltedUsers.js
#### How should this be manually tested?
Create accounts in Canvas and CTL online with the same email address (I've just been using my own)
Modify account on CTLonline to be a trainee (have that 'Role')
Melt in a Canvas course (either by 'concluding Enrollment' https://crisistextline.instructure.com/courses/${courseID}/conclude_user/${userID} or Deactive the user in the course from GUI https://crisistextline.instructure.com/courses/${courseID}/users

Run script (Keys, Config, you know) 

'Role' in CTL online should be modified to 'Deferred' on melting from one course and 'Dropout' when melted for a second time. Emails should be sent. 

#### Any background context you want to provide?
#### What are the relevant tickets?
https://admin.crisistextline.org/jira/secure/RapidBoard.jspa?rapidView=5&view=detail&selectedIssue=INT-472
#### Questions:
Bluebird is already required by the request module. Do we need to require it again in the main package.json?
